infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LayoutSwitchDao.kt

```diff
@@ -434,4 +434,24 @@ class LayoutSwitchDao(jdbcTemplateParam: NamedParameterJdbcTemplate?) :
             )
         }
     }
+
+    fun officialDuplicateNameExistsFor(switchId: IntId<TrackLayoutSwitch>): Boolean {
+        val sql = """
+            select
+              exists(
+                  select *
+                    from layout.switch this_switch
+                      join layout.switch duplicate_switch
+                           on this_switch.name = duplicate_switch.name and this_switch.id != duplicate_switch.id
+                    where not duplicate_switch.draft
+                      and duplicate_switch.state_category != 'NOT_EXISTING'
+                      and this_switch.id = :switchId)
+        """.trimIndent()
```

Comment:
Mmh, mietin että riittääkö näissä pelkkä officialin tarkastelu... Voisi ehkä olla kiva, että jos operaattori koittaa nimetä vaihdetta (tai raidetta, tai jotain muuta assettia) samalle nimelle kuin joku olemassaoleva, niin siitä varoitettaisiin vaikka molemmat olisivat vasta drafteja.

Myöskään koko reviewiä en ole vielä lukenut (fast trackasin tänne kun näin nämä `officialDuplicateNameExistsFor`-nimet), mutta onko se case otettu jotenkin erillisenä huomioon, että koitan luoda vaikka 2 raidetta samalla nimellä, ja julkaisen ne molemmat samassa julkaisussa?